modal/serving.py

```diff
@@ -97,6 +105,8 @@ async def _run_watch_loop(
 async def _serve_app(
     app: "_App",
     app_ref: str,
+    *,
+    is_module: bool,
```

Comment:
Do we consider `serve_app` to be public API? It looks a lot like the `App.run` and `modal.runner.deploy_app` API analogues for the core CLI, so it feels like we should expose it just for completeness sake. OTOH it's a little less obvious how it would be useful to use it programmatically.

It bears on whether we want "internal" parameters like `is_module` to appear in the signature. But see comment above about whether we can reduce the need to be passing `is_module` around.

Reply:
I don't think we should make it public since it's quite clunky at the moment in having to specify the "app_ref" which is a very cli thing imo

Reply:
Yeah, it's because of how the serve command works when files change ("auto reloading") - it starts a new interpreter that should reload everything based on the original command, so it needs the app ref for that reason :(